proj/hps_accel/vizier/client_vizier.py

```diff
@@ -0,0 +1,188 @@
+import math
+import random
+import time
+
+from google.cloud import storage
+from googleapiclient import discovery, errors
+from scipy.fftpack import idct
+
+from vizier.client import OptimizerClient
+from vizier.vapi import *
+
+_OPTIMIZER_API_DOCUMENT_BUCKET = 'caip-optimizer-public'
+_OPTIMIZER_API_DOCUMENT_FILE = 'api/ml_public_google_rest_v1.json'
+
+random.seed()
+
+def _read_api_document(project_id):
+    client = storage.Client(project_id)
+    bucket = client.get_bucket(_OPTIMIZER_API_DOCUMENT_BUCKET)
+    blob = bucket.get_blob(_OPTIMIZER_API_DOCUMENT_FILE)
+    return blob.download_as_string()
+
+def _trial_name(project_id, region, study_id, trial_id):
+  return f'projects/{project_id}/locations/{region}/studies/{study_id}/trials/{trial_id}'
+
+def trial_from_gtrial(study: Study, id, gtrial):    
+    trial = Trial(
+        client_id = id,
+        state = gtrial['state'],
+        parameters = [],
+        measurements = []
+    )
+
+    for param in gtrial['parameters']:
+        pname = param['parameter']
+        ptype = study.parameters[pname].type
+        pval = None
+        if ptype in ['DOUBLE', 'DISCRETE']:
+            pval = param['floatValue']
+        elif ptype == 'INTEGER':
+            pval = param['intValue']
+        elif ptype == 'CATEGORICAL':
+            pval = param['stringValue']
+        else:
+            raise Exception('Unknown parameter type')
+
+        p: Parameter = study.parameters[pname].trial(pname, pval)
+        trial.parameters += [p]
+    
+    return trial
+
+class VizierClient(OptimizerClient):
```

Comment:
did you consider using https://googleapis.dev/python/aiplatform/latest/aiplatform_v1beta1/vizier_service.html? that might allow you to get rid of the helper classes.

Reply:
I've mostly followed the examples from here, which do not use that: https://cloud.google.com/ai-platform/optimizer/docs

The `OptimizerClient` class that `VizierClient` is built on top is an abstraction layer that allows other back-ends, such as Advisor, to be used in place of Vizier, but implementations of their respective clients is not included within this PR.

Reply:
I see, it's nice that the implementation is swappable but I don't think that prevent us from reusing existing high level client to implemented the plumbing with vizier.

I also found https://www.tensorflow.org/cloud/guides/cloud_tuner_guide which seems to provide an even higher-level interface (and is also used in the official samples: https://cloud.google.com/ai-platform/optimizer/docs/optimizing-multiple-objectives)

Reply:
There might also be an deeper way to integrate this into the Vertex AI platform using: https://cloud.google.com/vertex-ai/docs/training/using-hyperparameter-tuning

They basically have a high level [method](https://googleapis.dev/python/aiplatform/latest/aiplatform_v1/job_service.html?highlight=create_hyperparameter_tuning_job#google.cloud.aiplatform_v1.services.job_service.JobServiceClient.create_hyperparameter_tuning_job) that allow to create a full hyperparameter search pipeline and point it to a python entrypoint that takes the parameters as command line arguments and report metrics by writing them to disk, see https://cloud.google.com/vertex-ai/docs/training/code-requirements#write_code_for_hyperparameter_tuning

The nice thing is that it allow you to visualize the progress the Vertex AI UI:
![image](https://user-images.githubusercontent.com/5268/155620512-ce78d3bb-3482-4503-ab11-7b928c1cf8d8.png)

I'm not sure how partical it is for non-AI workflow, but I'd be happy to give it a try if you think that helps.